src/openstudio_lib/VariablesTabView.cpp

```diff
@@ -183,8 +265,45 @@ void VariablesList::allOffClicked() {
 }
 
 void VariablesList::enableAll(bool t_enabled) {
-  for (auto itr = m_variables.begin(); itr != m_variables.end(); ++itr) {
-    (*itr)->setVariableEnabled(t_enabled);
+  for (auto& m_variable : m_variables) {
+    if (m_variable->isVisible()) {
+      m_variable->setVariableEnabled(t_enabled);
+    }
+  }
+}
```

Comment:
@macumber Here is where the "All On" button only affects Visible variables already.

Reply:
Oh cool, ok thanks